### ISSUE TITLE
Scenario where mutliple accounts with the same identifier are returned by pca.getAccounts()

### ISSUE THREAD

**Issue by dan-trewin** [stashenergy/react-native-msal#80]
My enterprise has many apps that use adal/msal (which all share the same cache on device).
Our app uses the `"account_mode": "MULTIPLE"` setting in the `msal_config.json` as it's required for this library, but we really only ever want one account signed into the app.

@josmithua here https://github.com/stashenergy/react-native-msal/issues/38#issuecomment-692285044, you described using a js wrapper to "only allow a sign in if there are not any existing accounts".
We tried to achieve this by checking the length of `pca.getAccounts()`. This is our pseudocode:

```js
const accounts = await pca.getAccounts()

if (accounts.length === 0) {
   // If no accounts are currently logged in, prompt for creds via the embedded web view or the authenticator app
   return await getTokenInteractively(scopes)
} else if (accounts.length === 1) {
   // If one account is currently logged in, get their token silently
   return await getTokenSilently(accounts[0], scopes)
} else {
   // If multiple accounts are currently logged in, sign them all out
   console.log('Multiple accounts detected. This is not allowed. Logging them all out.')

   accounts.forEach(async (account) => {
      await logOut(account)
   })

   return await getTokenInteractively(scopes)
}
```

This has worked great for us on ios (and android emulators), but when we used this solution and put the apk on a physical android device, we ran into an occasional issue where the auth is infinitely asking for an interactive login. This is b/c 3 accounts were being returned from pca.getAccounts() that were all identical and all had the same account identifier. When we try to call pca.removeAccount(account) on all 3 of them on that "foreach log out loop", it will remove the account the first time and subsequent attempts will error with the following adb log: 
```
MsalOAuth2TokenCache:getAccount:  [2021-04-23 16:14:45 - {"thread_id":"9433","correlation_id":"4d36d9a8-d6b9-4f11-9dff-54d93db62676"}] No matching account found. Android 29

MsalOAuth2TokenCache:removeAccount:  [2021-04-23 16:14:45 - {"thread_id":"9433","correlation_id":"4d36d9a8-d6b9-4f11-9dff-54d93db62676"}] Insufficient filtering provided for account removal - preserving Account. Android 29
```
This is fine b/c we catch the logout error and ignore it. However, on the very next `await getTokenInteractively(scopes)` call (`pca.getToken(scopes)`), we sign in once and again get 3 accounts signed in all at once. Now our app will try to log out all those accounts next time you try to get a token (b/c we only ever want 1 account signed in) hence the infinite loop.

Honestly I am not sure how to reproduce this issue. I believe it only happens when using an intune wrapped apk with a broker (ours is company portal), and even then it is rare. I also don't know if it is an issue with this library or the underlying native android library (microsoft-authentication-library-for-android).

Any idea why pca.getAccounts would be returning 3 identical accounts? Any suggestions are greatly appreciated. The only solution I can think of at the moment is to filter out the extra accounts from pca.getAccounts if/when the account identifiers are the same.

 - `react-native-msal` version: 3.1.0
 - `react-native` version: 0.63.4
 - Platform: Android

**Comment by dan-trewin** [comment]
I do think if I un-enrolled my device from intune and re-enrolled it, it would also fix the issue for the particular device I'm using, but I'm trying to find a solution that doesn't require that. If this happened to one of our users it would be a pain.

**Comment by josmithua** [comment]
Hi Dan,

I've ran into this issue once before on an Android emulator. I tried logging out of the device, rebooting, reinstalling the app, etc., but the only thing that worked was to delete the emulator and create a new one... Then the duplicate accounts were gone, and I haven't had an issue after that. This happened when we made some breaking changes to our Azure tenant. I haven't been able to reproduce the issue since.

I chalked it up to the inconsistency that occurred when I was logged in to the emulator and then we made the changes to our tenant and then tried using the app again.

Also, the accounts may have the same identifier, but if you inspect the whole object, are they completely identical? Some properties might be different, which may give you another clue. 

I'm really sorry I can't help you more right now, just don't have enough info.

**Comment by dan-trewin** [comment]
Hi Josh,

I haven't ever had the issue on an android emulator. It's only ever happened for us on a physical android device. There didn't seem to be any way to resolve it. I tried uninstalling and reinstalling the app through our enterprise google play store. The phone I was testing with continued to get multiple identical accounts. When we did have this issue, I did inspect the whole account object. They were in fact completely identical.

Thanks for taking the time to respond. It's a very tricky issue. I do believe the problem is most likely with some wonky setup with our test Azure tenant (kind of like you mentioned).

To be sure this issue won't happen in prod, we opted for filtered out the extra accounts from pca.getAccounts in javascript code if/when the account identifiers are the same. Our users haven't reported any issues so far. 

I'm okay with closing out this issue.
